### DISCUSSION TITLE
jupyter-book failing in _toc

### DISCUSSION THREAD

**Discussion by chipbrock** [executablebooks/meta#389] [Q&A]
Am a newbie. A refugee from Bookdown trying to move to Jupyter-book. I have been happy running my own and others' tutorials for days now, but all of the sudden I've done something that causes the build to fail in a strange way. The basic issue appears to be at:

`Running Jupyter-Book v0.10.2
Source Folder: /Users/me/jupyter_books/chipbook
Config Path: /Users/me/jupyter_books/chipbook/_config.yml
Output Path: /Users/me/jupyter_books/chipbook/_build/html
Running Sphinx v3.5.4
WARNING: Unknown key in `_toc.yml`: format
WARNING: Unknown key in `_toc.yml`: root
WARNING: Unknown key in `_toc.yml`: options`

my toc and config files are attached (converting .yml extension to .txt so they would upload).

[_config.txt](https://github.com/executablebooks/meta/files/6620097/_config.txt)
[_toc.txt](https://github.com/executablebooks/meta/files/6620098/_toc.txt)

I'd sure appreciate some advice... This is obviously global since some of the books that were working yesterday have not been touched but all three throw the same errors.

**Comment by chrisjsewell** [comment]
Heya, Your toc format, is the new format that only works from jupyter-book v0.11, but above you are using jupyter-book v0.10

**Reply by chipbrock** [reply]
Ah. Appreciate the quick response. You know one figures stuff out by looking at docs and tutorials. Clearly this project is moving fast and stuff is mixing. I've tried multiple ways to reinstall and I think update, but that's not working. what's the best way to command-line stay current with Jupyter Books?  thanks

**Reply by choldgraf** [reply]
`pip install -U jupyter-book`

**Reply by chipbrock** [reply]
Thanks. I tried that yesterday and it didn't work. So I've done it again with the original _toc.yml which is:

```
`# Table of content
# Learn more at https://jupyterbook.org/customize/toc.html
#
format: jb-book
root: intro
options:
  numbered: true
sections:
- file: the_planet
- file: foundations/L-051_motion_g_galileo2
- file: randomly/content
- file: randomly/mymarkdownfile
- file: randomly/notebooks`

I get the toc malformed error... 

`Running Jupyter-Book v0.11.1
Traceback (most recent call last):
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/jupyter_book/cli/main.py", line 243, in build
    parse_toc_yaml(toc)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/sphinx_external_toc/parsing.py", line 82, in parse_toc_yaml
    return parse_toc_data(data)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/sphinx_external_toc/parsing.py", line 100, in parse_toc_data
    doc_item, docs_list = _parse_doc_item(
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/sphinx_external_toc/parsing.py", line 146, in _parse_doc_item
    raise MalformedError(
sphinx_external_toc.parsing.MalformedError: Unknown keys found: {'sections'}, allowed: {'format', 'root', 'options', 'title', 'meta', 'defaults', 'parts', 'chapters'} @ '/'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/chipbrock/opt/anaconda3/bin/jupyter-book", line 8, in <module>
    sys.exit(main())
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/jupyter_book/cli/main.py", line 245, in build
    _error(
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/jupyter_book/utils.py", line 48, in _error
    raise kind(box)
RuntimeError: 
===============================================================================

The Table of Contents file is malformed: Unknown keys found: {'sections'}, allowed: {'format', 'root', 'options', 'title', 'meta', 'defaults', 'parts', 'chapters'} @ '/'
You may need to migrate from the old format, using:

	jupyter-book toc migrate /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_toc.yml -o /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_toc.yml

===============================================================================`
```
 If I go back to a really simple, but I gather old _toc with just - file: X lines, it works.

**Reply by choldgraf** [reply]
per your error message, you have `sections` in your TOC file and this is not allowed, it needs to be `chapters`

this is the important part of the error message:

```
**The Table of Contents file is malformed: Unknown keys found: {'sections'}, allowed: {'format', 'root', 'options', 'title', 'meta', 'defaults', 'parts', 'chapters'}
```

**Reply by chipbrock** [reply]
You're right. Thanks. It works now.

This has been a tail-chasing event. That upgrade, while obviously useful for the future, has been sort of hard to follow in the ramifications. It would be really useful for folks like me who are starting and maybe straddling between two versions, to have a "here's what worked before" and "here's what works now"  The 

`jupyter-book toc migrate /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_toc.yml -o /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_toc.yml`

command I now see the purpose of. But it took that old version _toc and added the format and root preamble which I didn't realize were new things.

Anyhow. I sound stupid. thanks for your help. I'm now back to what started all of this - trying to use my own css, which was in my original post before I broke the toc!
